src/components/Stories/TrendingStoryDetail.js

```diff
@@ -22,12 +22,33 @@ class TrendingStoryDetail extends Component {
     super(props);
     this.handleUpVoteClick = this.handleUpVoteClick.bind(this);
     this.handleDownVoteClick = this.handleDownVoteClick.bind(this);
+    this.approveVote = this.approveVote.bind(this);
   }
 
   contentHtml(data) {
     return ReactHtmlParser(data);
   }
 
+  approveVote(response, story) {
+    var arr = this.props.TrendingStoriesReducer.approvedStories;
+    var indexOfId = _.findIndex(arr, { id: story.id });
+    arr.splice(indexOfId, 1, response.data);
+
+    let keyDown = story.id + 'up';
```

Comment:
If the user can only either upvote or downvote a story, we should probably store the voted story id in a set in `localStorage` for quick lookup

Reply:
But, in my opinion we should not prevent user from up or down voting. We need to decrease
the count of the other vote when user clicks on one.

Reply:
sure, i think site like reddit also does that

Reply:
I made the alteration to just 1 local variable. I, too, agree that the user should be able to change their mind fluidly. I propose we leave that for the last sprint, for touch-up